src/ResourceManager/Resources/Commands.Resources/RoleAssignments/NewAzureRoleAssignmentCommand.cs

```diff
@@ -146,6 +146,10 @@ public class NewAzureRoleAssignmentCommand : ResourcesBaseCmdlet
         [ValidateGuidNotEmpty]
         public Guid RoleDefinitionId { get; set; }
 
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Delegation flag.")]
```

Comment:
@darshanhs90 nit: switch parameters shouldn't have the `ValueFromPipelineByPropertyName` property